handlers/filter.go

```diff
@@ -0,0 +1,81 @@
+package handlers
+
+import (
+	"context"
+	"errors"
+	"net/url"
+	"strings"
+
+	search "github.com/ONSdigital/dp-api-clients-go/site-search"
+	"github.com/ONSdigital/dp-frontend-search-controller/data"
+	"github.com/ONSdigital/log.go/log"
+)
+
+func mapFilterTypes(ctx context.Context, query url.Values) (apiQuery url.Values, err error) {
+	apiQuery, err = url.ParseQuery(query.Encode())
+	if err != nil {
+		log.Event(ctx, "failed to parse copy of query for mapping filter types", log.Error(err), log.ERROR)
+		return nil, err
+	}
+	filters := apiQuery["filter"]
+	if len(filters) > 0 {
+		var newFilters []string
+		for _, fType := range filters {
+			found := false
+			for _, category := range data.Category {
+				for _, searchType := range category {
+					if fType == searchType.QueryType {
+						found = true
+						newFilters = append(newFilters, searchType.SubTypes)
+						break
```

Comment:
This is awesome. I have never seen this before and I like it. It is so much neater and logical rather than doing multiple breaks. I have implemented it so that the loop is named